torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -156,11 +156,20 @@ void insertUniquePredicated(
 }
 
 void SegmentedGroup::finalize() {
-  // Move all the edgees to group input/output
+  // Move all the edges to group input/output
   // Inputs
   insertUniquePredicated(
       input_vals, producer_edges, [](Val* v) { return !v->isFusionInput(); });
 
+  for (auto expr : exprs_) {
+    for (auto i : expr->inputs()) {
+      if (i->isAnInt() && i->definition() == nullptr && !i->isConstScalar() &&
+          !i->isFusionInput()) {
```

Comment:
dumb question, are we skipping `isFusionInput()` because fusion inputs are available by default to every segmented group?

Reply:
nvm, I saw the section below that seems to be doing that with tensor_map already.

Reply:
Yes. The segmented groups would know it if the scalars are input. This fix is just for the scalars that don't know where they came from. One example is extents of `TensorView`. Currently the tv's know where to find it's extents but unfortunately there's not a way to specify a clear `definition()` for these scalars.